units/host_status.go

```diff
@@ -226,7 +226,22 @@ func (j *cloudHostReadyJob) setNextState(h host.Host) error {
 		// can just set it as running.
 		return errors.Wrap(h.MarkAsProvisioned(), "marking host as running")
 	default:
-		return errors.Wrap(h.SetProvisioning(), "marking host as provisioning")
+		// All other host types must be manually provisioned by the app server.
+		if err := h.SetProvisioning(); err != nil {
+			return errors.Wrap(err, "marking host as provisioning")
+		}
+
+		setupJob := NewSetupHostJob(j.env, h, utility.RoundPartOfMinute(0).Format(TSFormat))
```

Comment:
Should id be higher than a minute? Since host provisioning is likely to take several minutes, isn't it likely that we will get duplicate jobs from the cron job running every minute?

Reply:
It doesn't matter because the scopes are applied on enqueue. This code most likely will only run once for a given host, and even if it runs more than once, it won't enqueue the duplicate job because of the scope.